consensus/polybft/fsm.go

```diff
@@ -517,13 +517,11 @@ func (f *fsm) verifyValidatorsUptimeTx(transactions []*types.Transaction) error
 				createdUptimeTx.Hash,
 			)
 		}
-	} else {
-		if blockUptimeTx != nil && blockUptimeTx.Hash == createdUptimeTx.Hash {
-			return errors.New("didn't expect uptime transaction in the middle of an epoch")
```

Comment:
This error message indicates that this was already checked, wasn't this working as expected?

Reply:
We didn't need this condition `blockUptimeTx != nil && blockUptimeTx.Hash == createdUptimeTx.Hash`. If we managed to decode state transaction, and that transaction was uptime, if it wasn't an end of epoch, we should always return this error.